src/dms/frontend/EdFi.DataManagementService.Frontend.AspNetCore.Tests.Unit/Modules/DiscoveryModuleTests.cs

```diff
@@ -68,4 +69,64 @@ public async Task Discovery_Endpoint_Returns_Ok_Response()
         apiDetails?.dataModels.Count().Should().Be(1);
         apiDetails?.dataModels[0].name.Should().Be("Ed-Fi");
     }
+
+    [Test]
+    public async Task When_PathBase_Provided_Discovery_Endpoint_Returns_Ok_Response()
+    {
+        // Arrange
+        var versionProvider = A.Fake<IVersionProvider>();
+        A.CallTo(() => versionProvider.Version).Returns("1.0");
+        A.CallTo(() => versionProvider.ApplicationName).Returns("DMS");
+
+        IDataModelInfo expectedDataModelInfo = (
+            new
+            {
+                ProjectName = "Ed-Fi",
+                ProjectVersion = "5.0.0",
+                Description = "Ed-Fi data standard 5.0.0",
+            }
+        ).ActLike<IDataModelInfo>();
+        var apiService = A.Fake<IApiService>();
+        A.CallTo(() => apiService.GetDataModelInfo()).Returns([expectedDataModelInfo]);
+
+        var pathBase = "dms-api";
+        await using var factory = new WebApplicationFactory<Program>().WithWebHostBuilder(builder =>
+        {
+            builder.UseEnvironment("Test");
+            builder.ConfigureAppConfiguration(
+                (context, configuration) =>
+                {
+                    configuration.AddInMemoryCollection(
+                        new Dictionary<string, string?> { ["AppSettings:PathBase"] = pathBase }
+                    );
+                }
+            );
+            builder.ConfigureServices(
+                (collection) =>
+                {
+                    collection.AddTransient((x) => versionProvider);
+                    collection.AddTransient((x) => apiService);
+                }
+            );
+        });
+        using var client = factory.CreateClient();
+
+        // Act
+        var response = await client.GetAsync($"/{pathBase}");
+        var content = await response.Content.ReadAsStringAsync();
+        var apiDetails = JsonSerializer.Deserialize<DiscoveryApiDetails>(content);
+
+        // Assert
+        response.StatusCode.Should().Be(HttpStatusCode.OK);
+        apiDetails.Should().NotBeNull();
+        apiDetails?.urls.Count.Should().Be(5);
+        var dependenciesUrl = apiDetails?.urls.First(x => x.Key.Equals("dependencies"));
+        dependenciesUrl.Should().NotBeNull();
+        dependenciesUrl?.Value.Should().NotBeNull();
+        dependenciesUrl!.Value!.ToString().Should().Contain(pathBase);
+        apiDetails?.applicationName.Should().Be("DMS");
+        apiDetails?.dataModels.Should().NotBeNull();
+        apiDetails?.dataModels.Count().Should().Be(1);
+        apiDetails?.dataModels[0].name.Should().Be("Ed-Fi");
+    }
```

Comment:
Can you add a test that POST response `Location` header correctly includes the base path?
Want to ensure that `AspNetCoreFrontend.ToResult()` sets the correct Location header from `frontendResponse.LocationHeaderPath`. It might need to be an E2E test.